packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1513 @@
+        messageType: AgreementAddedV1,
+        payload: actualAgreementCreatedEvent.data,
+      }).agreement;
+
+      const expectedCreatedAgreement = toAgreementV1({
+        id: newAgreementId,
+        eserviceId: agreementToBeUpgraded.eserviceId,
+        descriptorId,
+        producerId: agreementToBeUpgraded.producerId,
+        consumerId: agreementToBeUpgraded.consumerId,
+        verifiedAttributes: agreementToBeUpgraded.verifiedAttributes,
+        certifiedAttributes: agreementToBeUpgraded.certifiedAttributes,
+        declaredAttributes: agreementToBeUpgraded.declaredAttributes,
+        consumerNotes: agreementToBeUpgraded.consumerNotes,
+        state: agreementState.draft,
+        createdAt: TEST_EXECUTION_DATE,
+        consumerDocuments: [],
+        stamps: {},
+      });
+
+      expectedCreatedAgreement.stamps = {};
+      delete expectedCreatedAgreement.updatedAt;
+      delete expectedCreatedAgreement.suspendedAt;
+      delete expectedCreatedAgreement.contract;
+      expect(actualCreatedAgreement).toMatchObject(expectedCreatedAgreement);
+
+      // The following assertions verify the "AgreementConsumerDocumentAdded" event for each document
+      for (let index = 0; index < agreementConsumerDocuments.length; index++) {
+        const agreementConsumerDocument = agreementConsumerDocuments[index];
+        const currentVersion = index + 1;
+        const actualAgreementDocumentAddedEvent =
+          await readEventByStreamIdAndVersion(
+            newAgreementId,
+            currentVersion,
+            "agreement",
+            postgresDB
+          );
+
+        expect(actualAgreementDocumentAddedEvent).toMatchObject({
+          type: "AgreementConsumerDocumentAdded",
+          event_version: 1,
+          version: currentVersion.toString(),
+          stream_id: newAgreementId,
+        });
+
+        const actualAgreementDocumentAdded = decodeProtobufPayload({
+          messageType: AgreementConsumerDocumentAddedV1,
+          payload: actualAgreementDocumentAddedEvent.data,
+        }).document;
+
+        expect(actualAgreementDocumentAdded).toBeDefined();
+        if (!actualAgreementDocumentAdded) {
+          fail("Document not found in event");
+        }
+        const expectedUploadedDocumentPath = `${config.consumerDocumentsPath}/${newAgreementId}/${actualAgreementDocumentAdded.id}/${agreementConsumerDocument.name}`;
+
+        const expectedCreatedDocument = {
+          id: unsafeBrandId<AgreementDocumentId>(
+            actualAgreementDocumentAdded.id
+          ),
+          name: agreementConsumerDocument.name,
+          prettyName: agreementConsumerDocument.prettyName,
+          contentType: agreementConsumerDocument.contentType,
+          path: expectedUploadedDocumentPath,
+          createdAt: TEST_EXECUTION_DATE,
+        };
+
+        expect(actualAgreementDocumentAdded).toMatchObject(
+          toAgreementDocumentV1(expectedCreatedDocument)
+        );
```

Comment:
As discussed, here you could also add: 

```ts
expect(await fileManager.listFiles(config.s3Bucket)).toContain(
    expectedUploadedDocumentPath
);
```

And you could add this expect also in all other tests where you clone documents